src/FromJSONTrait.php

```diff
@@ -27,9 +27,38 @@ public static function fromJSON($jsonStr) {
 
         if (is_null($cfg)) {
             $err = json_last_error();
-            throw new \InvalidArgumentException("Invalid JSON: $err");
+            throw new JsonParseErrorException($cfg, $err, self::provideJsonErrorMessageForCode($err));
         }
         $called_class = get_called_class();
         return new $called_class($cfg);
     }
+
+    private static function provideJsonErrorMessageForCode($code)
+    {
+
+        /**
+         * This is a hastily thrown together fix for the fact that
+         * php 5.4 doesn't have json_last_error_msg(). I very much recommend
+         * that this is replaced with json_last_error_msg(), is removed and the minimum
+         * recommended version of php is changed to 5.5. Especially seeing as php 5.4
+         * is now no longer being supported as of me writing this.
+         * @see http://php.net/supported-versions.php
+         */
```

Comment:
Yep the comment can go. With regards to versions upping these to 5.6.* should be fine given that 5.5 won't be receiving security updates in 5 months time.

0.12.0 will still support 5.5 and 5.4.